### PR TITLE
TDR 3239 - Make consignmentId option filter for getting metadata

### PR DESCRIPTION
**Pull request by tna-aford**
[TDR-3239](https://national-archives.atlassian.net/browse/TDR-3239?atlOrigin=eyJpIjoiMDA0MTY3N2JjYTFmNDNmYmEzY2VlMTQ0MjFhNDc4ZjkiLCJwIjoiaiJ9)

- [x] Method altered
- [x] Tests updated

[TDR-3239]: https://national-archives.atlassian.net/browse/TDR-3239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala [hunk 1]
```diff
@@ -127,7 +127,7 @@ class FileMetadataService(
   }
 
   def getFileMetadata(consignmentId: UUID, selectedFileIds: Option[Set[UUID]] = None): Future[Map[UUID, FileMetadataValues]] =
```
Comment (vimleshtna) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala:129]: I think you need to make `consignmentId` argument an `Option` too and pass `None` from the `getPaginatedFiles()` method. https://github.com/nationalarchives/tdr-consignment-api/blob/master/src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala#L165C43-L165C58

`fileMetadata <- fileMetadataService.getFileMetadata(consignmentId, fileIds)`

**Review by vimleshtna — COMMENTED** [review]
Looks good, just a minor comment. thanks.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ValidateFileMetadataService.scala [hunk 2]
```diff
@@ -37,7 +37,7 @@ class ValidateFileMetadataService(
   def validateAdditionalMetadata(fileIds: Set[UUID], consignmentId: UUID, propertiesToValidate: Set[String]): Future[List[FilestatusRow]] = {
     for {
       customMetadataFields <- customMetadataService.getCustomMetadata
-      existingMetadataProperties: Seq[FilemetadataRow] <- fileMetadataRepository.getFileMetadata(consignmentId, Some(fileIds), Some(toPropertyNames(customMetadataFields)))
+      existingMetadataProperties: Seq[FilemetadataRow] <- fileMetadataRepository.getFileMetadata(Some(consignmentId), Some(fileIds), Some(toPropertyNames(customMetadataFields)))
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/ValidateFileMetadataService.scala:40]: I don't think the consignment id needs to be passed in here, as we're using the file ids.

**Review by TomJKing — COMMENTED** [review]
@tna-aford @vimleshtna Added an additional minor comment. Thanks

**Review by vimleshtna — APPROVED** [review]
Looks good, thanks

**Pull request merged**